controllers/restic.go

```diff
@@ -68,11 +70,22 @@ func getNodeAgentObjectMeta(r *DPAReconciler) metav1.ObjectMeta {
 func (r *DPAReconciler) ReconcileResticDaemonset(log logr.Logger) (bool, error) {
 	// When updating from OADP version 1.1.x to 1.2.x, delete
 	// daemonset.apps/restic (it was renamed to daemonset.apps/node-agent)
+	// resticrepositories.velero.io CRD (it was renamed to backuprepositories.velero.io)
 	if err := r.Delete(r.Context, &appsV1.DaemonSet{ObjectMeta: metaV1.ObjectMeta{
 		Name: "restic", Namespace: r.NamespacedName.Namespace,
 	}}); err != nil && !apiErrors.IsNotFound(err) {
 		return false, err
 	}
+	resticRepositoriesResource := &unstructured.Unstructured{}
```

Comment:
2 questions:
- right place? this only gets created (instances) in 1.1 if user enables restic? But even if user does not create an instance, the CRD will always be there
- should we delete instances prior to deleting CRD?

Reply:
Deleting CRD will automatically delete instances as long as there's no finalizers.

Reply:
> this only gets created (instances) in 1.1 if user enables restic?

Yes.

> even if user does not create an instance, the CRD will always be there

Yes. CRD installation handled by OLM, and is defined by what CRD is in the `bundle/` and so it is not dynamically created when needed by oadp reconciliation of dpa.

Reply:
I will check if there are finalizers

Reply:
about location I was worried that if user creates DPA without restic, it won't trigger this function, but it was triggered, right?

Reply:
This [runs](https://github.com/openshift/oadp-operator/blob/b570aa4012b9acce7e48bdd6c13ab4b9bdd7cd02/controllers/dpa_controller.go#L102) regardless of dpa settings.

Reply:
@mateusoliveira43 Checking for finalizers means listing and iterating over every ResticRepository in the namespace. That may be overkill, since Velero never adds finalizers to these CRs.

Reply:
oh I meant manually deleting CRD in my cluster with instances, and see if any problem occurs hehehehe Just to check if I do not have to add delete instances code

Reply:
> Checking for finalizers means listing and iterating over every ResticRepository

Actually I think it would show up in CRD status in some form that there exists CRs with finalizers.. so we can work off that rather than querying ourselves each item.

Reply:
@kaovilai I don't think that's true. For example, velero Restores have a finalizer, but if you get the restore CRD, there's nothing in status about restore CRs or their finalizers.

Reply:
> velero Restores have a finalizer, but if you get the restore CRD, there's nothing in status about restore CRs or their finalizers.

I don't mean this scenario. I meant during deletion.

See [example CRD status here](https://stackoverflow.com/questions/52009124/not-able-to-completely-remove-kubernetes-customresource#:~:text=Finally%2C%20here%20is%20the%20output%20of%2C)

```
Status:
  Conditions:
    Last Transition Time:  2018-08-23T13:29:45Z
    Message:               CustomResource deletion is in progress
    Reason:                InstanceDeletionInProgress
    Status:                True
    Type:                  Terminating
Events:                    <none>
```